hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -688,6 +688,10 @@ private BiConsumer<Fetchable, Boolean> createFetchableBiConsumer(
 			LoaderSqlAstCreationState creationState,
 			ImmutableFetchList.Builder fetches) {
 		return (fetchable, isKeyFetchable) -> {
+			if ( !fetchable.isSelectable() ) {
```

Comment:
I'm curious now. When you add that here, `BaseSqmToSqlAstConverter` is the only place where we still process the fetchable even though it is not selectable. Should we add the same condition there as well? What if we simply remove the fetchable from the `staticFetchableList` instead?